shared/js/places.js

```diff
@@ -0,0 +1,103 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+/**
+  'places' DataStore wrapper
+  Ensures the data format of the 'places' DataStore stays consistent.
+**/
+
+'use strict';
+
+/* exported getPlacesStore */
+
+var getPlacesStore = (() => {
+  var _store;
+  var verify = (obj) => {
+    var field;
+    var obligatory = {
+      url: 'string'
+    };
+    var allowed = {
+      url: 'string',
+      title: 'string',
+      icons: 'object',
+      frecency: 'number',
+      visits: 'object', // Array, to be specific
+      screenshot: 'object',
+      fxsyncId: 'string',
+      visited: 'number'
+    };
+    if (typeof obj !== 'object') {
+      throw new Error(`Records in places DataStore should be objects.`);
+    }
+    for (field in obligatory) {
+      if (typeof obj[field] !== obligatory[field]) {
+        throw new Error(`Missing obligatory field ${field} in places DataStore \
```

Comment:
We dont use this type of wrapping in general and just in terms of aesthetics would prefer to avoid, concatting seems nicer
```
      throw new Error(`Missing obligatory field ${field} in places DataStore` + 
                                `record`);
```
So the new line aligns with the rest, of if possible dropping some of the text so its < 80 chars